spec/requests/providers_spec.rb

```diff
@@ -1156,6 +1156,55 @@ def gen_import_request
     end
   end
 
+  describe 'change provider password' do
+    let(:ems_physical_infra) { FactoryGirl.create(:ems_physical_infra) }
+    let(:invalid_change_password_payload) do
+      { "action"           => "change_password",
+        "current_password" => "current_password",
+        "new_password"     => "new_password",
+        "confirm_password" => "confirm_password" }
+    end
+
+    let(:valid_change_password_payload) do
+      { "action"           => "change_password",
+        "current_password" => "current_password",
+        "new_password"     => "new_password",
+        "confirm_password" => "new_password" }
+    end
+
+    context 'with confirmation password different of new_password' do
+      it 'returns a bad request error' do
+        api_basic_authorize collection_action_identifier(:providers, :change_password)
+
+        post(api_provider_url(nil, ems_physical_infra), :params => invalid_change_password_payload)
+
+        expect_bad_request("Confirm password did not match.")
+      end
+    end
+
+    context 'with a non existent provider' do
+      it 'returns a not found error' do
+        api_basic_authorize collection_action_identifier(:providers, :change_password)
+
+        post(api_provider_url(nil, 999_999), :params => valid_change_password_payload)
+
+        expect(response).to have_http_status(:not_found)
+      end
+    end
+
+    context 'with a valid request' do
+      it 'proccess the request successfully' do
+        allow_any_instance_of(ExtManagementSystem).to receive(:change_password) { true }
```

Comment:
Would it be possible to do this without `any_instance_of`? Have you looked into using webmock or something similar? Since this is an integration test, stubbing it at the network layer would give me greater confidence that everything is wired up correctly. And, since I had some concerns about all providers responding to this (above), that would give me greater confidence as a passer-by that we haven't stubbed a method that doesn't exist.

Reply:
To me, this is a big code smell - we've stubbed a method that doesn't exist on the target object. This makes me question the value of this test - is it just there to prove that the stub works?

This ties into https://github.com/ManageIQ/manageiq-api/pull/309#discussion_r182776204 - we can't smooth over those details. I want to see in another test what happens when the provider doesn't implement password change. As for this test, I would much rather that a) the specific lenovo provider is built for this test, and b) we stub at the network layer